src/Framework/Support/Scripts/Concerns/HasScriptAssetFile.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Give\Framework\Support\Scripts\Concerns;
+
+trait HasScriptAssetFile {
```

Comment:
I'm tempted to say this would make a better `Assets` facade, rather than modifying the class the assets are used within. We're really just using this to make these functions available, as it's not even adding any properties, and a facade is better suited for that when we're not really trying to compose anything specific.

Reply:
Agreed, I updated with a new `ScriptAsset` facade here 42a79ab1

Since this is so specific to the dependency extraction and generated "asset" file, I made this specific to the script asset files hence the `ScriptAsset` name.